### PR TITLE
Add and use AP_GRIPPER_ENABLED

### PR DESCRIPTION
**Pull request by peterbarker**
This tidies many vehicle's existing `GRIPPER_ENABLED` define so we also remove random library functions which are gripper-related.

### CODE REVIEW

File: libraries/AP_Gripper/AP_Gripper.cpp [hunk 1]
```diff
@@ -109,12 +111,16 @@ void AP_Gripper::init()
     switch(config.type.get()) {
     case 0:
         break;
+#if AP_GRIPPER_SERVO_ENABLED
     case 1:
         backend = new AP_Gripper_Servo(config);
         break;
+#endif
+#if AP_GRIPPER_SERVO_ENABLED
```
Comment (amilcarlucas) [inline libraries/AP_Gripper/AP_Gripper.cpp:119]:
```suggestion
#if AP_GRIPPER_EPM_ENABLED
```

**Review by amilcarlucas — COMMENTED** [review]
I wonder how much it saves on small boards

File: libraries/AP_Gripper/AP_Gripper.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_Gripper/AP_Gripper.cpp:119]: Fixed, thanks

**Comment by Hwurzburg** [comment]
remember any option we change we need to change the build_options.py and rebuild the features.txt, right?

**Comment by peterbarker** [comment]
> remember any option we change we need to change the build_options.py and rebuild the features.txt, right?

Yep.  I'll be setting a cronjob up tomorrow to update the server every hour or something.

**Comment by Hwurzburg** [comment]
dont think it needs to be every hour if its a load issue....every day or so should do if it is

**Comment by tridge** [comment]
need to work out why Pixhawk1-1M saves flash space with this change

**Comment by peterbarker** [comment]
> need to work out why Pixhawk1-1M saves flash space with this change

Previously gripper was removed at the vehicle level, but the compiler was unable to remove the library functions.  Now they're explicitly cut out based on the define:
image
image

**Review by tridge — APPROVED** [review]

**Pull request merged**